### PR TITLE
MODLOGIN-99

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
Implement reset password action API:
 * Add implementation "/authn/password-reset-action"
 * Add rest tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -366,6 +427,7 @@
     <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (kostyantyn-kh) [inline pom.xml]: could you move <properties> section to the top for better readability?

File: pom.xml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (kostyantyn-kh) [inline pom.xml]: version should be changed as well

File: ramls/login.raml [hunk 3]
```diff
@@ -245,5 +249,60 @@ traits:
...
                     example: "Internal server error"
-
-
+  /reset-password:
+    post:
+      description: Resets password for user in record and deletes action record
+      body:
+        application/json:
+          schema: passwordReset
+      responses:
```
Comment (kostyantyn-kh) [inline ramls/login.raml]: add 400(Bad request)

File: ramls/login.raml [hunk 4]
```diff
@@ -245,5 +249,60 @@ traits:
...
+          body:
+            text/plain:
+              example: "Bad Request"
+        500:
+          description: "Internal server error"
+          body:
+            text/plain:
+              example: "Internal server error"
+    /{actionId}:
+      get:
```
Comment (kostyantyn-kh) [inline ramls/login.raml]: add 400(Bad request)

File: ramls/login.raml (hunk 3 above)
Comment (OleksiiMaksymov) [reply ramls/login.raml]: Fix

**Comment by OleksiiMaksymov** [comment]
Thanks for the comments, all comments taken into account.

**Comment by OleksiiKuzminov** [comment]
plz make lint-raml happy and add descriptions

File: descriptors/ModuleDescriptor-template.json [hunk 5]
```diff
@@ -52,6 +52,21 @@
           "methods": [ "POST" ],
           "pathPattern" : "/authn/password/repeatable",
           "permissionsRequired" : [ "login.password.validate" ]
+        },
```
Comment (OleksiiKuzminov) [inline descriptors/ModuleDescriptor-template.json]: Update interface version "version": "4.3"
You add new endpoints so should update api version

File: src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+        .get(SNAPSHOTS_TABLE_PW, PasswordCreate.class, criterion, true, false,
+          getReply ->
+          {
+            if (getReply.failed()) {
+              asyncHandler.handle(Future.succeededFuture(null));
+              return;
+            }
+            Optional<PasswordCreate> passwordCreateOpt = getReply.result().getResults().stream().findFirst();
+            if (!passwordCreateOpt.isPresent()) {
+              asyncHandler.handle(Future.succeededFuture(null));
```
Comment (OleksiiKuzminov) [inline src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java]: why null?

File: descriptors/ModuleDescriptor-template.json (hunk 5 above)
Comment (OleksiiMaksymov) [reply descriptors/ModuleDescriptor-template.json]: Thanks for the comments!
Done

File: src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java (hunk 6 above)
Comment (OleksiiMaksymov) [reply src/main/java/org/folio/services/impl/PasswordStorageServiceImpl.java]: Made changes to the implementation

File: NEWS.md [hunk 7]
```diff
@@ -1,3 +1,15 @@
+## 2018-11-19 v4.5.0
+ * Add implementation /authn/password-reset-action which is accessible through OKAPI
+ * Add rest tests
+
+ API:
+
+ | METHOD |  URL                                    | DESCRIPTION                                                     |
+ |--------|-----------------------------------------|-----------------------------------------------------------------|
+ | POST   | /authn/reset-password                   | Saves action to storage                                         |
```
Comment (opochepam) [inline NEWS.md]: swap description with /authn/password-reset-action endpoint

File: descriptors/ModuleDescriptor-template.json [hunk 8]
```diff
@@ -52,6 +52,21 @@
...
+        {
+          "methods": [ "GET" ],
+          "pathPattern" : "/authn/password-reset-action/{actionId}",
+          "permissionsRequired" : [ "login.password-reset-action.get" ]
+        },
+        {
+          "methods": [ "POST" ],
+          "pathPattern" : "/authn/reset-password",
+          "permissionsRequired" : [ "login.password-reset-action.post" ]
```
Comment (opochepam) [inline descriptors/ModuleDescriptor-template.json]: wrong permission name

Comment (opochepam) [reply descriptors/ModuleDescriptor-template.json]: should be smth like login.password-reset.post

Comment (OleksiiMaksymov) [reply descriptors/ModuleDescriptor-template.json]: Thanks!
Done

**Pull request merged**